source/server/transport_socket_config_impl.h

```diff
@@ -37,14 +36,22 @@ class TransportSocketFactoryContextImpl : public TransportSocketFactoryContext {
 
   Stats::Store& stats() override { return stats_; }
 
+  void setInitManager(Init::Manager& init_manager) override {
+    init_manager_ = std::unique_ptr<Init::Manager>(&init_manager);
```

Comment:
You shouldn't take ownership of init manager here, just set it to a raw pointer, or use std::reference_wrapper